### ISSUE TITLE
[BUG]: Handle File Path '\\' under Windows

### ISSUE THREAD

**Issue by linrongbin16** [utilyre/barbecue.nvim#63]
### Requirements

- [X] I have scammed [other issues](https://github.com/utilyre/barbecue.nvim/issues) to see if this bug has already been reported.
- [X] I'm fully certain that this is not a breaking change, see [the latest release](https://github.com/utilyre/barbecue.nvim/releases/latest).
- [X] This plugin and its dependencies are up to date with the latest commit.
- [X] I'm willing to fix this through a PR.

### Expected Behavior

In windows cmd/powershell, file path is not separated by '/', but it's a '\\'.
Expect this plugin split path correctly under windows.

### Actual Behavior

The file path splits not actually working:
![image](https://user-images.githubusercontent.com/6496887/217293746-b89e0513-6b4c-4047-b295-10292137a083.png)

### Neovim Version

```plain text
latest stable: 0.8.2
```

### Minimal Configuration

```Lua
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  -- do not remove the colorscheme!
  "folke/tokyonight.nvim",

  {
    "utilyre/barbecue.nvim",
    dependencies = {
      "neovim/nvim-lspconfig",
      "SmiteshP/nvim-navic",
      "nvim-tree/nvim-web-devicons",
    },
    config = function() 
      require("barbecue").setup()
    end,
  },
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

-- add anything else here
vim.opt.termguicolors = true
-- do not remove the colorscheme!
vim.cmd([[colorscheme tokyonight]])
```

### Reproduction

use lazy.nvim to install barbecue and neo-tree.nvim, open this plugin project and will see it.

**Comment by linrongbin16** [comment]
I'm trying to working on a PR to fix it.

**Comment by linrongbin16** [comment]
This issue is fixed in #64
